lib/web_ui/lib/src/engine/view_embedder/style_manager.dart

```diff
@@ -78,106 +78,86 @@ void applyGlobalCssRulesToSheet(
   String cssSelectorPrefix = '',
   required String defaultCssFont,
 }) {
-  // TODO(web): use more efficient CSS selectors; descendant selectors are slow.
-  // More info: https://csswizardry.com/2011/09/writing-efficient-css-selectors
-
-  assert(styleElement.sheet != null);
-  final DomCSSStyleSheet sheet = styleElement.sheet! as DomCSSStyleSheet;
-
-  // Fixes #115216 by ensuring that our parameters only affect the flt-scene-host children.
-  sheet.insertRule('''
-    $cssSelectorPrefix ${DomManager.sceneHostTagName} {
-      font: $defaultCssFont;
-    }
-  ''', sheet.cssRules.length);
+  styleElement.appendText(
```

Comment:
This might be a violation of the CSP of the page. Have you tried to run an app with this change with the following meta-tags?

```
script-src 'nonce-YOUR_NONCE_VALUE' 'wasm-unsafe-eval';
font-src https://fonts.gstatic.com;
style-src 'nonce-YOUR_NONCE_VALUE';
```

(Remember to set the `nonce` values in your flutter.js initialization, as described [here](https://github.com/flutter/flutter/pull/137204))

Reply:
It seems to be working fine in a sample app that I tried (I had to `nonce` in many places to get it working 😜).

@ditman were you concerned that `Element.appendText(...)` is not allowed with CSP? Now that I think about it, how is it different from the `insertRule` API? They both take an arbitrary string.

Reply:
The difference is that `insertRule` is inserting the rule in CSSOM (so there's no possibility of XSS, AFAIK), but `insertText` is appending stuff in the DOM, so I thought that either TrustedTypes or CSP wouldn't like it!

Reply:
> I guess it's more like `textContent` than `innerHTML` 😅

Exactly. If you look at `appendText` in `dom.dart`, it's basically creating a Text Node and appending it. (and then, I assume, the CSS parser will parse it the same way it parses the string passed to `insertRule`).

Reply:
Just passing by but I believe XSS is possible in css too

Reply:
@cedvdb I this case I think this is all right, because the stylesheet that is being created and injected is signed with flutter's `nonce`.